prediction_market_agent/agents/think_thoroughly_agent/think_thoroughly_agent.py

```diff
@@ -383,8 +400,13 @@ def generate_prediction_for_one_outcome(
             list[tuple[str, AnswerWithScenario]] | None
         ) = None,
     ) -> AnswerWithScenario | None:
-        researcher = ThinkThoroughlyWithItsOwnResearch._get_researcher(model)
-        predictor = ThinkThoroughlyWithItsOwnResearch._get_predictor(model)
+        # Do not enable add_langfuse_callback, because it's not thread-safe.
```

Comment:
This is unfortunate, `generate_prediction_for_one_outcome` is executed in parallel, so we can't just decorate it with observe.  But we do track at least inputs and outputs.